backend/src/GraphQl/OpenIdConnect/Application/ApplicationType.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using HotChocolate;
+using HotChocolate.Types;
+using Metabase.Authorization;
+using Metabase.Data;
+using Metabase.GraphQl.Users;
+using Microsoft.AspNetCore.Identity;
+using OpenIddict.Core;
+
+namespace Metabase.GraphQl.OpenIdConnect.Application;
+
+public sealed class ApplicationType
+    : ObjectType<OpenIdApplication>
+{
+    protected override void Configure(
+        IObjectTypeDescriptor<OpenIdApplication> descriptor
+    )
+    {
+        const string suffixedName = nameof(ApplicationType);
+        descriptor.Name(suffixedName.Remove(suffixedName.Length - "Type".Length));
+
+        descriptor.Field(t => t.Permissions).Type<NonNullType<ListType<StringType>>>().Resolve(context =>
+        {
+            var application = context.Parent<OpenIdApplication>();
+            var permissions = JsonSerializer.Deserialize<List<string>>(application.Permissions!);
+            return permissions?.FindAll(permission => permission.Contains("api")).ToList();
+        });
+
+        descriptor.Field(t => t.RedirectUris).Type<NonNullType<StringType>>().Resolve(context =>
```

Comment:
Because you return only one URI, this field should have the singular name `RedirectUri`. The same naming change should be done for the other fields below.